### DISCUSSION TITLE
Why my marp-cli doesnt work in git-bash, but go well in powershell?

### DISCUSSION THREAD

**Discussion by Do7and** [marp-team/marp#500] [Q&A] [answered]
![image](https://github.com/marp-team/marp/assets/41378140/fe825904-99c6-4c38-bd29-b47b72866cc5)
![image](https://github.com/marp-team/marp/assets/41378140/968b3edd-b059-4a78-a110-16a0ad1d8e85)

When I type marp in bash, it get stuck and has no output even I use Ctrl+C to kill it. But when runs in powershell it works well.
I dont know why they are different. What should I do to find it out?
My marp is installed by npm.

**Comment by yhatt** [comment, marked as answer]
It looks similar to the case of `python`.
https://stackoverflow.com/questions/32597209/python-not-working-in-the-command-line-of-git-bash

In MinGW (Git Bash), Marp CLI would work by disabling to read piped stdin through `--no-stdin` option.

```bash
marp --no-stdin
```

Or use [`winpty`](https://github.com/rprichard/winpty) bridge to run Marp CLI.

```bash
# Use `marp.cmd` instead of `marp` when using Marp CLI installed by npm. winpty seems not to be able to automatically detect the correct extension for running.
winpty marp.cmd
```

**Reply by Do7and** [reply]
![image](https://github.com/marp-team/marp/assets/41378140/adb3dc3b-9d2c-4498-928f-531f0e5e715e)
Both can work! Thanks for your anwser!
